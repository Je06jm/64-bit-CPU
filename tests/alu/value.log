VCD info: dumpfile tests/alu/value.vcd opened for output.
VCD warning: $dumpvars: Package ($unit) is not dumpable with VCD.
VCD warning: $dumpvars: Package (instructions) is not dumpable with VCD.
VCD warning: $dumpvars: Package (types) is not dumpable with VCD.
Passed (tests/alu/value.sv, 36 [1]): 8 bit zero extend
Passed (tests/alu/value.sv, 40 [2]): 8 bit sign extend
Passed (tests/alu/value.sv, 44 [3]): 8 bit sign extend 2
Passed (tests/alu/value.sv, 50 [4]): 16 bit zero extend
Passed (tests/alu/value.sv, 54 [5]): 16 bit sign extend
Passed (tests/alu/value.sv, 58 [6]): 16 bit sign extend 2
Passed (tests/alu/value.sv, 64 [7]): 32 bit zero extend
Passed (tests/alu/value.sv, 68 [8]): 32 bit sign extend
Passed (tests/alu/value.sv, 72 [9]): 32 bit sign extend 2
Passed (tests/alu/value.sv, 78 [10]): 64 bit zero extend
Passed (tests/alu/value.sv, 82 [11]): 64 bit sign extend
Passed (tests/alu/value.sv, 86 [12]): 64 bit sign extend 2
tests/alu/value.sv:89: $finish called at 12 (1s)
