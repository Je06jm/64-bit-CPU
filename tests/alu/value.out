#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1110-g18392a46)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2009.vpi";
S_0000022911afdfb0 .scope package, "$unit" "$unit" 2 1;
 .timescale 0 0;
S_0000022911afe140 .scope package, "instructions" "instructions" 3 7;
 .timescale 0 0;
enum0000022911bc7e10 .enum4 (16)
   "NOP" 16'b0000000000000000,
   "HALT" 16'b0000000000000001,
   "LOADMCR" 16'b0000000000000010,
   "STOREMCR" 16'b0000000000000011,
   "SYSCALL" 16'b0000000000000100,
   "SYSRET" 16'b0000000000000101,
   "INT" 16'b0000000000000110,
   "INTRET" 16'b0000000000000111,
   "ADD" 16'b0000000000001000,
   "SUB" 16'b0000000000001001,
   "MUL" 16'b0000000000001010,
   "UDIV" 16'b0000000000001011,
   "UMOD" 16'b0000000000001100,
   "SDIV" 16'b0000000000001101,
   "SMOD" 16'b0000000000001110,
   "INC" 16'b0000000000001111,
   "DEC" 16'b0000000000010000,
   "FADD" 16'b0000000000010001,
   "FSUB" 16'b0000000000010010,
   "FMUL" 16'b0000000000010011,
   "FDIV" 16'b0000000000010100,
   "FMOD" 16'b0000000000010101,
   "FFMA" 16'b0000000000010110,
   "FEPOW" 16'b0000000000010111,
   "F2POW" 16'b0000000000011000,
   "F10POW" 16'b0000000000011001,
   "FEPOW1" 16'b0000000000011010,
   "F2POW1" 16'b0000000000011011,
   "F10POW1" 16'b0000000000011100,
   "FLN" 16'b0000000000011101,
   "F2LOG" 16'b0000000000011110,
   "F10LOG" 16'b0000000000011111,
   "FLN1" 16'b0000000000100000,
   "F2LOG1" 16'b0000000000100001,
   "F10LOG1" 16'b0000000000100010,
   "FDISTANCE" 16'b0000000000100011,
   "FSQRT" 16'b0000000000100100,
   "FPOW1" 16'b0000000000100101,
   "FFPOW" 16'b0000000000100110,
   "FPOW" 16'b0000000000100111,
   "FSIN" 16'b0000000000101000,
   "FCOS" 16'b0000000000101001,
   "FTAN" 16'b0000000000101010,
   "FASIN" 16'b0000000000101011,
   "FACOS" 16'b0000000000101100,
   "FATAN" 16'b0000000000101101,
   "FATAN2" 16'b0000000000101110,
   "FSINPI" 16'b0000000000101111,
   "FCOSPI" 16'b0000000000110000,
   "FTANPI" 16'b0000000000110001,
   "FASINPI" 16'b0000000000110010,
   "FACOSPI" 16'b0000000000110011,
   "FATANPI" 16'b0000000000110100,
   "FATAN2PI" 16'b0000000000110101,
   "FSINH" 16'b0000000000110110,
   "FCOSH" 16'b0000000000110111,
   "FTANH" 16'b0000000000111000,
   "FARSINH" 16'b0000000000111001,
   "FARCOSH" 16'b0000000000111010,
   "FARTANH" 16'b0000000000111011,
   "FABS" 16'b0000000000111100,
   "FNEG" 16'b0000000000111101,
   "FCOPY" 16'b0000000000111110,
   "FTEST" 16'b0000000000111111,
   "TOINT" 16'b0000000001000000,
   "TOFLOAT" 16'b0000000001000001,
   "AND" 16'b0000000001000010,
   "OR" 16'b0000000001000011,
   "XOR" 16'b0000000001000100,
   "NOT" 16'b0000000001000101,
   "ROLR" 16'b0000000001000110,
   "ROLL" 16'b0000000001000111,
   "SHIFTR" 16'b0000000001001000,
   "SHIFTL" 16'b0000000001001001,
   "FLIP" 16'b0000000001001010,
   "SWAP" 16'b0000000001001011,
   "MOVE" 16'b0000000001001100,
   "CMP" 16'b0000000001001101,
   "JMP" 16'b0000000001001110,
   "BRA" 16'b0000000001001111,
   "RET" 16'b0000000001010000,
   "LOAD" 16'b0000000001010001,
   "STORE" 16'b0000000001010010,
   "STOREIF" 16'b0000000001010011,
   "PUSH" 16'b0000000001010100,
   "POP" 16'b0000000001010101,
   "BIT" 16'b0000000001010110,
   "OUT" 16'b0000000001010111,
   "IN" 16'b0000000001011000,
   "SETF" 16'b0000000001011001,
   "CLEARF" 16'b0000000001011010
 ;
enum0000022911bcb8a0 .enum4 (4)
   "BITS_8" 4'b0000,
   "BITS_16" 4'b0001,
   "BITS_32" 4'b0010,
   "BITS_64" 4'b0011
 ;
S_0000022911ac9780 .scope package, "types" "types" 4 4;
 .timescale 0 0;
enum0000022911bc7d60 .enum4 (2)
   "HALF" 2'b00,
   "SINGLE" 2'b01,
   "DOUBLE" 2'b10
 ;
S_0000022911ac9910 .scope module, "Arithmetic" "Arithmetic" 5 7;
 .timescale 0 0;
    .port_info 0 /INPUT 16 "op";
    .port_info 1 /INPUT 1 "carryIn";
    .port_info 2 /INPUT 64 "a";
    .port_info 3 /INPUT 64 "b";
    .port_info 4 /OUTPUT 64 "result";
    .port_info 5 /OUTPUT 1 "divByZero";
    .port_info 6 /OUTPUT 1 "zero";
    .port_info 7 /OUTPUT 1 "carry";
    .port_info 8 /OUTPUT 1 "negitive";
o0000022911b1a558 .functor BUFZ 64, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0000022911afbcb0_0 .net "a", 63 0, o0000022911b1a558;  0 drivers
o0000022911b1a588 .functor BUFZ 64, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0000022911ad8e90_0 .net "b", 63 0, o0000022911b1a588;  0 drivers
v0000022911ae3250_0 .var "carry", 0 0;
o0000022911b1a5e8 .functor BUFZ 1, C4<z>; HiZ drive
v0000022911ae32f0_0 .net "carryIn", 0 0, o0000022911b1a5e8;  0 drivers
v0000022911ae3390_0 .var "divByZero", 0 0;
v0000022911ae3430_0 .var "negitive", 0 0;
o0000022911b1a678 .functor BUFZ 16, C4<zzzzzzzzzzzzzzzz>; HiZ drive
v0000022911ae34d0_0 .net "op", 15 0, o0000022911b1a678;  0 drivers
v0000022911ada4d0_0 .var "result", 63 0;
v0000022911ada570_0 .var/s "sa", 63 0;
v0000022911ada610_0 .var/s "sb", 63 0;
v0000022911ada6b0_0 .var "zero", 0 0;
E_0000022911b12c80/0 .event anyedge, v0000022911afbcb0_0, v0000022911ad8e90_0, v0000022911ada4d0_0, v0000022911ae34d0_0;
E_0000022911b12c80/1 .event anyedge, v0000022911ae32f0_0, v0000022911ada570_0, v0000022911ada610_0;
E_0000022911b12c80 .event/or E_0000022911b12c80/0, E_0000022911b12c80/1;
S_0000022911ac9aa0 .scope module, "ICompare" "ICompare" 6 5;
 .timescale 0 0;
    .port_info 0 /INPUT 64 "a";
    .port_info 1 /INPUT 64 "b";
    .port_info 2 /OUTPUT 1 "zero";
    .port_info 3 /OUTPUT 1 "negitive";
o0000022911b1a918 .functor BUFZ 64, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0000022911b71df0_0 .net "a", 63 0, o0000022911b1a918;  0 drivers
o0000022911b1a948 .functor BUFZ 64, C4<zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz>; HiZ drive
v0000022911b71a30_0 .net "b", 63 0, o0000022911b1a948;  0 drivers
v0000022911b72070_0 .var "negitive", 0 0;
v0000022911b71d50_0 .var "result", 63 0;
v0000022911b717b0_0 .var "zero", 0 0;
E_0000022911b123c0 .event anyedge, v0000022911b71df0_0, v0000022911b71a30_0, v0000022911b71d50_0;
S_0000022911ae30c0 .scope module, "IValueTest" "IValueTest" 7 9;
 .timescale 0 0;
v0000022911b71f30_0 .net "result", 63 0, v0000022911b71ad0_0;  1 drivers
v0000022911b71b70_0 .var "signExtend", 0 0;
v0000022911b71e90_0 .var "size", 3 0;
v0000022911b71cb0_0 .var "value", 63 0;
S_0000022911ada750 .scope module, "ivalue" "IValue" 7 15, 8 7 0, S_0000022911ae30c0;
 .timescale 0 0;
    .port_info 0 /INPUT 64 "in";
    .port_info 1 /OUTPUT 64 "out";
    .port_info 2 /INPUT 4 "size";
    .port_info 3 /INPUT 1 "signExtend";
v0000022911b718f0_0 .net "in", 63 0, v0000022911b71cb0_0;  1 drivers
v0000022911b71ad0_0 .var "out", 63 0;
v0000022911b721b0_0 .net "signExtend", 0 0, v0000022911b71b70_0;  1 drivers
v0000022911b71990_0 .net "size", 3 0, v0000022911b71e90_0;  1 drivers
E_0000022911b12440 .event anyedge, v0000022911b71990_0, v0000022911b721b0_0, v0000022911b718f0_0;
    .scope S_0000022911ac9910;
T_0 ;
    %wait E_0000022911b12c80;
    %load/vec4 v0000022911afbcb0_0;
    %assign/vec4 v0000022911ada570_0, 0;
    %load/vec4 v0000022911ad8e90_0;
    %assign/vec4 v0000022911ada610_0, 0;
    %load/vec4 v0000022911ada4d0_0;
    %pushi/vec4 0, 0, 64;
    %cmp/e;
    %flag_get/vec4 4;
    %assign/vec4 v0000022911ada6b0_0, 0;
    %load/vec4 v0000022911ada4d0_0;
    %parti/s 1, 63, 7;
    %assign/vec4 v0000022911ae3430_0, 0;
    %load/vec4 v0000022911ae34d0_0;
    %pushi/vec4 13, 0, 16;
    %cmp/e;
    %flag_get/vec4 4;
    %load/vec4 v0000022911ae34d0_0;
    %pushi/vec4 11, 0, 16;
    %cmp/e;
    %flag_get/vec4 4;
    %or;
    %load/vec4 v0000022911ae34d0_0;
    %pushi/vec4 14, 0, 16;
    %cmp/e;
    %flag_get/vec4 4;
    %or;
    %load/vec4 v0000022911ae34d0_0;
    %pushi/vec4 12, 0, 16;
    %cmp/e;
    %flag_get/vec4 4;
    %or;
    %load/vec4 v0000022911ad8e90_0;
    %pushi/vec4 0, 0, 64;
    %cmp/e;
    %flag_get/vec4 4;
    %and;
    %assign/vec4 v0000022911ae3390_0, 0;
    %load/vec4 v0000022911ae34d0_0;
    %dup/vec4;
    %pushi/vec4 8, 0, 16;
    %cmp/u;
    %jmp/1 T_0.0, 6;
    %dup/vec4;
    %pushi/vec4 9, 0, 16;
    %cmp/u;
    %jmp/1 T_0.1, 6;
    %dup/vec4;
    %pushi/vec4 10, 0, 16;
    %cmp/u;
    %jmp/1 T_0.2, 6;
    %dup/vec4;
    %pushi/vec4 11, 0, 16;
    %cmp/u;
    %jmp/1 T_0.3, 6;
    %dup/vec4;
    %pushi/vec4 12, 0, 16;
    %cmp/u;
    %jmp/1 T_0.4, 6;
    %dup/vec4;
    %pushi/vec4 13, 0, 16;
    %cmp/u;
    %jmp/1 T_0.5, 6;
    %dup/vec4;
    %pushi/vec4 14, 0, 16;
    %cmp/u;
    %jmp/1 T_0.6, 6;
    %dup/vec4;
    %pushi/vec4 15, 0, 16;
    %cmp/u;
    %jmp/1 T_0.7, 6;
    %dup/vec4;
    %pushi/vec4 16, 0, 16;
    %cmp/u;
    %jmp/1 T_0.8, 6;
    %jmp T_0.9;
T_0.0 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %load/vec4 v0000022911ad8e90_0;
    %pad/u 65;
    %add;
    %load/vec4 v0000022911ae32f0_0;
    %pad/u 65;
    %add;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.1 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %load/vec4 v0000022911ad8e90_0;
    %pad/u 65;
    %sub;
    %load/vec4 v0000022911ae32f0_0;
    %pad/u 65;
    %sub;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.2 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %load/vec4 v0000022911ad8e90_0;
    %pad/u 65;
    %mul;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.3 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %load/vec4 v0000022911ad8e90_0;
    %pad/u 65;
    %div;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.4 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %load/vec4 v0000022911ad8e90_0;
    %pad/u 65;
    %mod;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.5 ;
    %load/vec4 v0000022911ada570_0;
    %pad/s 65;
    %load/vec4 v0000022911ada610_0;
    %pad/s 65;
    %div/s;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.6 ;
    %load/vec4 v0000022911ada570_0;
    %pad/s 65;
    %load/vec4 v0000022911ada610_0;
    %pad/s 65;
    %mod/s;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.7 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %addi 1, 0, 65;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.8 ;
    %load/vec4 v0000022911afbcb0_0;
    %pad/u 65;
    %subi 1, 0, 65;
    %split/vec4 64;
    %assign/vec4 v0000022911ada4d0_0, 0;
    %assign/vec4 v0000022911ae3250_0, 0;
    %jmp T_0.9;
T_0.9 ;
    %pop/vec4 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0000022911ac9aa0;
T_1 ;
    %wait E_0000022911b123c0;
    %load/vec4 v0000022911b71df0_0;
    %load/vec4 v0000022911b71a30_0;
    %sub;
    %assign/vec4 v0000022911b71d50_0, 0;
    %load/vec4 v0000022911b71d50_0;
    %pushi/vec4 0, 0, 64;
    %cmp/e;
    %flag_get/vec4 4;
    %assign/vec4 v0000022911b717b0_0, 0;
    %load/vec4 v0000022911b71d50_0;
    %parti/s 1, 63, 7;
    %assign/vec4 v0000022911b72070_0, 0;
    %jmp T_1;
    .thread T_1, $push;
    .scope S_0000022911ada750;
T_2 ;
    %wait E_0000022911b12440;
    %load/vec4 v0000022911b71990_0;
    %dup/vec4;
    %pushi/vec4 0, 0, 4;
    %cmp/u;
    %jmp/1 T_2.0, 6;
    %dup/vec4;
    %pushi/vec4 1, 0, 4;
    %cmp/u;
    %jmp/1 T_2.1, 6;
    %dup/vec4;
    %pushi/vec4 2, 0, 4;
    %cmp/u;
    %jmp/1 T_2.2, 6;
    %dup/vec4;
    %pushi/vec4 3, 0, 4;
    %cmp/u;
    %jmp/1 T_2.3, 6;
    %jmp T_2.4;
T_2.0 ;
    %load/vec4 v0000022911b721b0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_2.5, 8;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 1, 7, 4;
    %replicate 56;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 8, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
    %jmp T_2.6;
T_2.5 ;
    %pushi/vec4 0, 0, 56;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 8, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
T_2.6 ;
    %jmp T_2.4;
T_2.1 ;
    %load/vec4 v0000022911b721b0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_2.7, 8;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 1, 15, 5;
    %replicate 48;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 16, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
    %jmp T_2.8;
T_2.7 ;
    %pushi/vec4 0, 0, 48;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 16, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
T_2.8 ;
    %jmp T_2.4;
T_2.2 ;
    %load/vec4 v0000022911b721b0_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_2.9, 8;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 1, 31, 6;
    %replicate 32;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 32, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
    %jmp T_2.10;
T_2.9 ;
    %pushi/vec4 0, 0, 32;
    %load/vec4 v0000022911b718f0_0;
    %parti/s 32, 0, 2;
    %concat/vec4; draw_concat_vec4
    %assign/vec4 v0000022911b71ad0_0, 0;
T_2.10 ;
    %jmp T_2.4;
T_2.3 ;
    %load/vec4 v0000022911b718f0_0;
    %assign/vec4 v0000022911b71ad0_0, 0;
    %jmp T_2.4;
T_2.4 ;
    %pop/vec4 1;
    %jmp T_2;
    .thread T_2, $push;
    .scope S_0000022911ae30c0;
T_3 ;
    %pushi/vec4 0, 0, 64;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000022911b71e90_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %end;
    .thread T_3;
    .scope S_0000022911ae30c0;
T_4 ;
    %vpi_call/w 7 29 "$dumpfile", "tests/alu/value.vcd" {0 0 0};
    %vpi_call/w 7 30 "$dumpvars" {0 0 0};
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000022911b71e90_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 255, 0, 64;
    %jmp/0xz  T_4.0, 4;
    %vpi_call/w 7 38 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000100100, $realtime, "8 bit zero extend" {0 0 0};
    %jmp T_4.1;
T_4.0 ;
    %vpi_call/w 7 40 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000100100, $realtime, "8 bit zero extend" {0 0 0};
T_4.1 ;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %cmp/e;
    %jmp/0xz  T_4.2, 4;
    %vpi_call/w 7 42 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000101000, $realtime, "8 bit sign extend" {0 0 0};
    %jmp T_4.3;
T_4.2 ;
    %vpi_call/w 7 44 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000101000, $realtime, "8 bit sign extend" {0 0 0};
T_4.3 ;
    %pushi/vec4 1, 0, 64;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 1, 0, 64;
    %jmp/0xz  T_4.4, 4;
    %vpi_call/w 7 46 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000101100, $realtime, "8 bit sign extend 2" {0 0 0};
    %jmp T_4.5;
T_4.4 ;
    %vpi_call/w 7 48 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000101100, $realtime, "8 bit sign extend 2" {0 0 0};
T_4.5 ;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0000022911b71e90_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 65535, 0, 64;
    %jmp/0xz  T_4.6, 4;
    %vpi_call/w 7 52 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000110010, $realtime, "16 bit zero extend" {0 0 0};
    %jmp T_4.7;
T_4.6 ;
    %vpi_call/w 7 54 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000110010, $realtime, "16 bit zero extend" {0 0 0};
T_4.7 ;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %cmp/e;
    %jmp/0xz  T_4.8, 4;
    %vpi_call/w 7 56 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000110110, $realtime, "16 bit sign extend" {0 0 0};
    %jmp T_4.9;
T_4.8 ;
    %vpi_call/w 7 58 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000110110, $realtime, "16 bit sign extend" {0 0 0};
T_4.9 ;
    %pushi/vec4 1, 0, 64;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 1, 0, 64;
    %jmp/0xz  T_4.10, 4;
    %vpi_call/w 7 60 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000111010, $realtime, "16 bit sign extend 2" {0 0 0};
    %jmp T_4.11;
T_4.10 ;
    %vpi_call/w 7 62 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000000111010, $realtime, "16 bit sign extend 2" {0 0 0};
T_4.11 ;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %pushi/vec4 2, 0, 4;
    %store/vec4 v0000022911b71e90_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 4294967295, 0, 64;
    %jmp/0xz  T_4.12, 4;
    %vpi_call/w 7 66 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001000000, $realtime, "32 bit zero extend" {0 0 0};
    %jmp T_4.13;
T_4.12 ;
    %vpi_call/w 7 68 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001000000, $realtime, "32 bit zero extend" {0 0 0};
T_4.13 ;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %cmp/e;
    %jmp/0xz  T_4.14, 4;
    %vpi_call/w 7 70 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001000100, $realtime, "32 bit sign extend" {0 0 0};
    %jmp T_4.15;
T_4.14 ;
    %vpi_call/w 7 72 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001000100, $realtime, "32 bit sign extend" {0 0 0};
T_4.15 ;
    %pushi/vec4 1, 0, 64;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 1, 0, 64;
    %jmp/0xz  T_4.16, 4;
    %vpi_call/w 7 74 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001001000, $realtime, "32 bit sign extend 2" {0 0 0};
    %jmp T_4.17;
T_4.16 ;
    %vpi_call/w 7 76 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001001000, $realtime, "32 bit sign extend 2" {0 0 0};
T_4.17 ;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %pushi/vec4 3, 0, 4;
    %store/vec4 v0000022911b71e90_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %cmp/e;
    %jmp/0xz  T_4.18, 4;
    %vpi_call/w 7 80 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001001110, $realtime, "64 bit zero extend" {0 0 0};
    %jmp T_4.19;
T_4.18 ;
    %vpi_call/w 7 82 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001001110, $realtime, "64 bit zero extend" {0 0 0};
T_4.19 ;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000022911b71b70_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %pushi/vec4 4294967295, 0, 32;
    %concati/vec4 4294967295, 0, 32;
    %cmp/e;
    %jmp/0xz  T_4.20, 4;
    %vpi_call/w 7 84 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001010010, $realtime, "64 bit sign extend" {0 0 0};
    %jmp T_4.21;
T_4.20 ;
    %vpi_call/w 7 86 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001010010, $realtime, "64 bit sign extend" {0 0 0};
T_4.21 ;
    %pushi/vec4 1, 0, 64;
    %store/vec4 v0000022911b71cb0_0, 0, 64;
    %delay 1, 0;
    %load/vec4 v0000022911b71f30_0;
    %cmpi/e 1, 0, 64;
    %jmp/0xz  T_4.22, 4;
    %vpi_call/w 7 88 "$display", "Passed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001010110, $realtime, "64 bit sign extend 2" {0 0 0};
    %jmp T_4.23;
T_4.22 ;
    %vpi_call/w 7 90 "$display", "Failed (%s, %0d [%0d]): %s", "tests/alu/value.sv", 32'sb00000000000000000000000001010110, $realtime, "64 bit sign extend 2" {0 0 0};
T_4.23 ;
    %vpi_call/w 7 89 "$finish" {0 0 0};
    %end;
    .thread T_4;
# The file index is used to find the file name in the following table.
:file_names 9;
    "N/A";
    "<interactive>";
    "-";
    "./rtl/instructions.svh";
    "./rtl/types.svh";
    "rtl/alu/arithmetic.sv";
    "rtl/alu/compare.sv";
    "tests/alu/value.sv";
    "rtl/alu/value.sv";
